In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.cross_validation import train_test_split

In [3]:
%matplotlib inline

In [5]:
spam = pd.read_csv("spambase.data")
spam.head()
spam.columns
isspam = spam[spam['1'] == 1]
notspam = spam[spam['1'] == 0]

###Lets see if we can tell the difference from spam and an actual email, with machine learning

In [6]:
spam_train, spam_test = train_test_split(spam, test_size = .40)

###Splitting our data for testing purposes ^^^^

In [7]:
spam_test
input_data_test = spam_test.ix[:,'0':"278"]
testing = spam_test[['1']]

In [8]:
input_data = spam_train.ix[:,'0':"278"]
training = spam_train[['1']]

####Lets try multinomial bayes and see how effectively it can tell the difference

In [9]:
from sklearn.naive_bayes import MultinomialNB
classifier = MultinomialNB()
classifier.fit(input_data, training)
classifier.score(input_data_test, testing)

/Users/BekkBlando/asd/lib/python3.4/site-packages/sklearn/utils/validation.py:449: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


0.78315217391304348

### Multinomial Bayes is able to predict determine if something is spam or not with a 78% accuracy